File: DevExpress-Demo/CS/MVVMDemo.Wpf/Modules/Services/NavigationService/CollectionViewModel.cs
using DevExpress.Mvvm;
using DevExpress.Mvvm.POCO;

namespace MVVMDemo.Services.Navigation {
    public class CollectionViewModel {
        INavigationService NavigationService { get { return this.GetRequiredService<INavigationService>(); } }

        public void ShowDetail(PersonInfo person) {
            if(person == null)
                return;
            NavigationService.Navigate(viewType: "NavigationDetailView", param: person, parentViewModel: this);
        }
    }
}

File: DevExpress-Demo/VB/MVVMDemo.Wpf/Modules/Services/NavigationService/CollectionViewModel.vb
Imports DevExpress.Mvvm
Imports DevExpress.Mvvm.POCO

Namespace MVVMDemo.Services.Navigation

    Public Class CollectionViewModel

        Private ReadOnly Property NavigationService As INavigationService
            Get
                Return GetRequiredService(Of INavigationService)()
            End Get
        End Property

        Public Sub ShowDetail(ByVal person As PersonInfo)
            If person Is Nothing Then Return
            NavigationService.Navigate(viewType:="NavigationDetailView", param:=person, parentViewModel:=Me)
        End Sub
    End Class
End Namespace
